sound/soc/sof/amd/acp-ipc.c

```diff
@@ -145,11 +148,21 @@ static void acp_dsp_ipc_get_reply(struct snd_sof_dev *sdev)
 irqreturn_t acp_sof_ipc_irq_thread(int irq, void *context)
 {
 	struct snd_sof_dev *sdev = context;
-	unsigned int dsp_msg_write = offsetof(struct scratch_ipc_conf, sof_dsp_msg_write);
-	unsigned int dsp_ack_write = offsetof(struct scratch_ipc_conf, sof_dsp_ack_write);
+	struct acp_dev_data *adata = sdev->pdata->hw_pdata;
+	unsigned int dsp_msg_write = sdev->debug_box.offset +
+				     offsetof(struct scratch_ipc_conf, sof_dsp_msg_write);
+	unsigned int dsp_ack_write = sdev->debug_box.offset +
+				     offsetof(struct scratch_ipc_conf, sof_dsp_ack_write);
 	bool ipc_irq = false;
 	int dsp_msg, dsp_ack;
 
+	if (adata->first_boot) {
+		snd_sof_ipc_msgs_rx(sdev);
+		acp_dsp_ipc_host_done(sdev);
+		adata->first_boot = false;
+		return IRQ_HANDLED;
+	}
```

Comment:
previously our inbox and outbox sizes are static fixed array sizes 0f 512. There we are not using window  information from firmware. https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/sof/ipc3.c#L629.

Now we are getting this info from firmware , and i am using that to my offsets.

to make backward compatibility ,what we done is in probe we are setting this boxes to 512. we don't have window info to get overridden. In this way we made it backward compatible

Reply:
Right, if the firmware has non default dsp/host box sizes (not 512) then the debug_box is moving and thus you are not going to be able to read the `sof_dsp_msg_write` since it is in a different position.
The dsp box offset is always 0, so you can read the FW_READY message just fine to extract the used window information.

But can you not use the existing `sdev->first_boot` flag?

Reply:
I checked this flag ..with in a fraction of seconds i am getting two interrupts  from firmware (before setting this flag sdev->first_boot =false)  ,and getting ipc errors.

Reply:
I see, so you are sending an IPC before sdev->first_boot is set to false and the reply is not handled.
`if (sdev->first_boot && sdev->fw_state != SOF_FW_BOOT_COMPLETE)`
should take care of that, I think.

Reply:
done

Reply:
It worked, Thanks for the Suggestion.